### PR TITLE
Update GPU installation guide to use CUDA 11.7

### PR DESCRIPTION
**Pull request by tonyhoo**
*Issue #, if available:*

[Issue 2885](https://github.com/autogluon/autogluon/issues/2885)

*Description of changes:*
Update GPU installation guide to use CUDA 11.7

* Log (from Linux with GPU)
```
dev-dsk-tonyhu-2c-77435e3d % pip install torch==1.13.1+cu117 torchvision==0.14.1+cu117 --extra-index-url https://download.pytorch.org/whl/cu117

pip install autogluon
Looking in indexes: https://pypi.org/simple, https://pypi.ngc.nvidia.com, https://download.pytorch.org/whl/cu117
Collecting torch==1.13.1+cu117
  Downloading https://download.pytorch.org/whl/cu117/torch-1.13.1%2Bcu117-cp39-cp39-linux_x86_64.whl (1801.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.8/1.8 GB 61.5 MB/s eta 0:00:00
Collecting torchvision==0.14.1+cu117
  Downloading https://download.pytorch.org/whl/cu117/torchvision-0.14.1%2Bcu117-cp39-cp39-linux_x86_64.whl (24.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 24.3/24.3 MB 55.4 MB/s eta 0:00:00
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CI/batch/docker/Dockerfile.gpu
Comment (yinweisu) [inline CI/batch/docker/Dockerfile.gpu:1]: why change from training to inference? Also this docker is used by our CI so should be built and push to our ECR before merging in

Comment (tonyhoo) [reply CI/batch/docker/Dockerfile.gpu:1]: Good catch, will switch back to training

Comment (tonyhoo) [reply CI/batch/docker/Dockerfile.gpu]: Talked with @yinweisu offline, agreed the update of image is not 0.8 relevant and I will revert this file change for now to unblock 0.8 release

**Review by yinweisu — APPROVED** [review]
LGTM! Feel free to merge once unit tests passed

**Comment by yinweisu** [comment]
Merge since CI for previous commit passed and changes shouldn't affect. The most recent failure should be due to aws outage

**Pull request merged**
